crates/sui-indexer-alt/src/ingestion/mod.rs

```diff
@@ -57,9 +62,18 @@ impl IngestionService {
         metrics: Arc<IndexerMetrics>,
         cancel: CancellationToken,
     ) -> Result<Self> {
+        // TODO: Potentially support a hybrid mode where we can fetch from both local and remote.
```

Comment:
Curious to hear more about this might work (what scenario you're thinking of).

Reply:
Local file sources are not as reliable as remote sources, so if we use local file source, it is always better to have a fallback remote source. For instance, if the FN just started up and is still warming up, or if the FN gets stuck or even crashes for whatever reason, we would want to keep the ingestion alive.

Reply:
Is this assuming the co-located `inotify` based approach? I had thought that was somewhat off the table for the near term.

Onboard for the idea of having a fallback though -- this is similar to what I had in mind for the remote ingestion set-up as well -- where the indexer would periodicially check with a fullnode whether it should switch to the fullnode's REST API for ingestion (if the fullnode is responsive and leading the indexer by a short amount), and would otherwise fallback to the ingestion service.